x/clp/keeper/grpc_query.go

```diff
@@ -84,6 +85,52 @@ func (k Querier) GetLiquidityProvider(c context.Context, req *types.LiquidityPro
 	return &lpResponse, nil
 }
 
+func (k Querier) GetLiquidityProviderData(c context.Context, req *types.LiquidityProviderDataReq) (*types.LiquidityProviderDataRes, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	if req.Pagination == nil {
+		req.Pagination = &query.PageRequest{
+			Limit: MaxPageLimit,
+		}
+	}
+
+	if req.Pagination.Limit > MaxPageLimit {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("page size greater than max %d", MaxPageLimit))
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+	addr, err := sdk.AccAddressFromBech32(req.LpAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	assetList, _, err := k.Keeper.GetAssetsForLiquidityProviderPaginated(ctx, addr, &query.PageRequest{Limit: MaxPageLimit})
```

Comment:
Why do we need to send the MaxPageLimit always ? 

Cant we , change it to 
`if req.Pagination.Limit > MaxPageLimit {
     req.Pagination.Limit = MaxPageLimit
}`

and then use req.Pagination.Limit here

Reply:
👍